sqlalchemy_jsonapi/declarative/serializer.py

```diff
@@ -0,0 +1,118 @@
+"""A serializer for serializing SQLAlchemy models to JSON API spec."""
+
+import datetime
+
+from inflection import dasherize, underscore
+
+
+class JSONAPISerializer(object):
+    """A JSON API serializer that serializes SQLAlchemy models."""
+
+    def serialize(self, resources):
+        """Serialize resource(s) according to json-api spec."""
+        try:
+            self.fields
+            self.model
+            self.dasherize
+        except AttributeError:
+            raise
+
+        if 'id' not in self.fields:
+            raise ValueError
```

Comment:
ValueError for what? We should have a message as part of the exception that explains why the exception was raised.